src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java

```diff
@@ -89,18 +89,21 @@ public String getServiceAccountJson() {
     return serviceAccountJson;
   }
 
+  @Nullable
   public String getServiceAccountType() {
-    if (containsMacro(NAME_SERVICE_ACCOUNT_TYPE) || Strings.isNullOrEmpty(serviceAccountType)) {
+    if (containsMacro(NAME_SERVICE_ACCOUNT_TYPE)) {
       return null;
     }
-    return serviceAccountType;
+    return Strings.isNullOrEmpty(serviceAccountType) ? SERVICE_ACCOUNT_FILE_PATH : serviceAccountType;
```

Comment:
Please add a comment in the javadoc explaining what is the default value that is interpreted